### PR TITLE
Bugfix closes #3

### LINKED ISSUE #3: Add bugfixes from the bugfix branch of the instructor's repository
Hi guys, yeah this is a good idea. I'm happy to add what I can.
You may be interested in checking out the [bugfix branch](https://github.com/bradtraversy/support-desk/tree/bugfix) of Brad's repository where I've been tracking and fixing many of the bugs found by students in the course, some from you @ArpadGBondor ! 
It's also been streamlined and simplified a good deal in terms of state management as there is no real need for a isLoading, isSuccess or reset state functions in tickets and notes. In most cases we can infer loading from presence or absence of data and leverage pending, fulfilled and rejected states provided by our AsyncThunkAction to reset state.
Also the useAuthStatus hook was removed as it was only really a wrapper for Redux state and only used in one place - PrivateRoute.jsx. 
PrivateRoute also got refactored to no longer use an `<Outlet />` as we were not really doing any nested routing so a `<Outlet />` was not necessary.
I haven't documented all the changes as yet but there are notes in the code where it has been changed and why.
I think the code from the bugfix branch would be a more stable and predictable base to start from for adding in additional features.
I'd be happy to submit a PR with the bug fixes if you like?

_Originally posted by @bushblade in https://github.com/ArpadGBondor/Brad_Traversy-React_Front_To_Back_2022-05-Support_Desk/issues/1#issuecomment-1191781771_

### PR DESCRIPTION
**Pull request by bushblade**
This PR will bring in the current changes in [Brad's bugfix branch.](https://github.com/bradtraversy/support-desk/tree/bugfix)
Closes #3 
Main changes would include.

- Remove unused vars
- Not sending a 200 response on errors thrown in API
- Fix app crashing on refresh in production - incorrect file path resolution
- Fix use of useEffect and missing dependencies in frontend
- Remove useAuthstatus hook - this was just a wrapper for Redux state and only used in PrivateRoute
- Remove Outlet from Private route as we are not using nested routing
- Remove unnecessary state and reset function from **auth** state, better leverage of AsyncThunkAction provided by Redux Toolkit
- Remove unnecessary state and reset function from **notes** state, better leverage of AsyncThunkAction provided by Redux Toolkit
- Remove unnecessary state and reset function from **tickets** state, better leverage of AsyncThunkAction provided by Redux Toolkit

I've also included a prettier config in the project that matches @ArpadGBondor 's coding style and formatting, so hopefully will maintain the style for any contributors using prettier as their formatter.

### CODE REVIEW

**Review by ArpadGBondor — APPROVED** [review]
Thank you! 😃

File: backend/middleware/errorMiddleware.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
-    const statusCode = res.statusCode || 500;
-    res.status(statusCode).json({
-        message: err.message,
-        stack: process.env.NODE_ENV === 'development' ? err.stack : '',
+const errorHandler = (error, _, res, next) => {
+    // FIX: check for bad status codes, if it's a good status code then we want to send
+    // a bad status code i.e. 2xx should not be sent as error response
+    const statusCode = res.statusCode < 400 ? 500 : res.statusCode;
+    console.log('error middleware');
+
```
Comment (ArpadGBondor) [inline backend/middleware/errorMiddleware.js:6]: delete console.log

File: frontend/src/features/tickets/ticketSlice.js [hunk 2]
```diff
@@ -59,36 +60,21 @@ export const ticketSlice = createSlice({
...
                 state.ticket = null;
             })
             .addCase(getTickets.fulfilled, (state, action) => {
                 state.tickets = action.payload;
-                state.isError = false;
-                state.message = '';
             })
             .addCase(getTickets.rejected, (state, action) => {
-                state.isError = true;
                 state.message = action.payload;
```
Comment (ArpadGBondor) [inline frontend/src/features/tickets/ticketSlice.js:72]: state.message doesn't exist at this point

**Review by ArpadGBondor — COMMENTED** [review]
I'm not trying to be too difficult, I'm just still trying to figure out how a code review should look like.

Most of these things are just cosmetics... 

Thank you very much for the PR Will.

**Comment by bushblade** [comment]
No problem at all, whatever makes sense to you is fine I think.
You're welcome.

**Pull request merged**